### PR TITLE
create tests for queue, all pass currently

### PR DESCRIPTION
**Pull request by AbrahamFergie**

### CODE REVIEW

File: spec/queue.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+
```
Comment (nicosesma) [inline spec/queue.js]: Tests look good it's great and interesting to use chai-change as well

File: src/queue.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export default class Queue {
+  constructor() {
+    this.head = 0
+    this.sequence = []
+  }
+
+  enqueue( data ) {
+    this.sequence[ this.head ] = data
+    this.head++
+  }
```
Comment (nicosesma) [inline src/queue.js]: i wonder if you could use 
```
 this.sequence[ this.head++ ] = data
```

File: src/queue.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+export default class Queue {
+  constructor() {
+    this.head = 0
```
Comment (nicosesma) [inline src/queue.js]: makes sense to have the counter, in order to keep track of the length

File: src/queue.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  dequeue() {
+    return this.head > 0 ? this.sequence.splice(0, 1)[0] : null
+  }
+
+  front() {
+    return this.head > 0 ? this.sequence[0] : null
+  }
+
+  back() {
+    return this.head > 0 ? this.sequence[ --this.head ] : null
```
Comment (nicosesma) [inline src/queue.js]: awesome

File: src/queue.js (hunk 2 above)
Comment (AbrahamFergie) [inline src/queue.js]: thanks man we did make that change

**Pull request merged**